### PR TITLE
Updates to the MYNN SFC layer and PBL schemes

### PR DESCRIPTION
**Pull request by ldfowler58**
We updated the MYNN Surface Layer and Planetary Boundary Layer (PBL) schemes. The PBL scheme includes the options needed to run the Eddy Diffusivity Mass Flux (EDMF) parameterizations.

* The core of the SFC Layer scheme is located in ./physics_mmm/sf_mynn.F.
*  The core of the PBL scheme is located in ./physics_mmm/bl_mynn.F and ./physics_mmm/bl_mynn_subroutines.F.

### CODE REVIEW

File: src/core_atmosphere/physics/physics_mmm/bl_mynn_subroutines.F [hunk 1]
```diff
@@ -0,0 +1,6565 @@
...
+        !  sm(k) = Prnum*sh(k)
+
+        !  !or blend them:
+        !  zi2   = MAX(zi, 300.)
+        !  wt    = .5*TANH((zw(k) - zi2)/200.) + .5
+        !  sm(k) = sm_pbl*(1.-wt) + sm(k)*wt
+       END IF !end Helfand & Labraga check
+
+       !Impose broad limits on Sh and Sm:
+       gmelq    = MAX(gmel/q3sq, 1e-8)
```
Comment (mgduda) [inline src/core_atmosphere/physics/physics_mmm/bl_mynn_subroutines.F:1815]: This line generates the following compilation error (with the `gnu` build target):
```
bl_mynn_subroutines.F:1815:33:

 1815 |        gmelq    = MAX(gmel/q3sq, 1e-8)
      |                                 1
Error: GNU Extension: Different type kinds at (1)
make[5]: *** [Makefile:53: bl_mynn_subroutines.o] Error 1
```
One possible resolution would be to change `1e-8` to `1d-8`.

**Review by mgduda — CHANGES_REQUESTED** [review]
Besides the compilation error for `bl_mynn_subroutines.F`, it looks like this PR branch pulls in commits from other branches (e.g., commits related to the `cu_ntiedtke_in` package in `Registry.xml`, so we should rebase this branch to include only MYNN changes.

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F [hunk 2]
```diff
@@ -651,37 +823,112 @@ subroutine driver_pbl(itimestep,configs,mesh,sfc_input,diag_physics,tend_physics
...
+       call mpas_log_write(' ')
+       call mpas_log_write('--- f_qc   = $l',logicArgs=(/f_qc/)  )
+       call mpas_log_write('--- f_qi   = $l',logicArgs=(/f_qi/)  )
+       call mpas_log_write('--- f_qs   = $l',logicArgs=(/f_qs/)  )
+       call mpas_log_write('--- f_qoz  = $l',logicArgs=(/f_qoz/) )
+       call mpas_log_write('--- f_nc   = $l',logicArgs=(/f_nc/)  )
+       call mpas_log_write('--- f_ni   = $l',logicArgs=(/f_ni/)  )
+       call mpas_log_write('--- f_nifa = $l',logicArgs=(/f_nifa/))
+       call mpas_log_write('--- f_nwfa = $l',logicArgs=(/f_nwfa/))
+       call mpas_log_write('--- f_nbca = $l',logicArgs=(/f_nbca/))
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F:872]: I'd propose removing these print statements, as they show up for every time step in the log file.

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F [hunk 3]
```diff
@@ -651,37 +823,112 @@ subroutine driver_pbl(itimestep,configs,mesh,sfc_input,diag_physics,tend_physics
...
+                  bl_mynn_cloudmix   = bl_mynn_cloudmix    ,                                           &
+                  bl_mynn_mixqt      = bl_mynn_mixqt       ,                                           &
+                  ids = ids , ide = ide , jds = jds , jde = jde , kds = kds , kde = kde ,              &
+                  ims = ims , ime = ime , jms = jms , jme = jme , kms = kms , kme = kme ,              &
+                  its = its , ite = ite , jts = jts , jte = jte , kts = kts , kte = kte ,              &
+                  errmsg = errmsg , errflg = errflg                                                    &
+                          )
+       call mpas_timer_stop('bl_mynn')
+       call mpas_log_write('--- exit subroutine mynn_bl_driver:')
+       call mpas_log_write(' ')
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F:931]: For these and similar print statements, can these be removed?

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F (hunk 2 above)
Comment (ldfowler58) [reply src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F]: As requested, I commented out the extra print statements. Although they add a little bit of clutter around the call to subroutine mynn_bl_driver, I thought that at least initially, it was important for users to know exactly which options are being used in the MYNN PBL scheme. Defaults values for the different options are of course in Registry.xml.

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F (hunk 3 above)
Comment (ldfowler58) [reply src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F]: As requested, I commented out the extra print statements.

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F [hunk 4]
```diff
@@ -653,37 +823,111 @@ subroutine driver_pbl(itimestep,configs,mesh,sfc_input,diag_physics,tend_physics
...
+       call mpas_log_write('--- f_qc   = $l',logicArgs=(/f_qc/)  )
+       call mpas_log_write('--- f_qi   = $l',logicArgs=(/f_qi/)  )
+       call mpas_log_write('--- f_qs   = $l',logicArgs=(/f_qs/)  )
+       call mpas_log_write('--- f_qoz  = $l',logicArgs=(/f_qoz/) )
+       call mpas_log_write('--- f_nc   = $l',logicArgs=(/f_nc/)  )
+       call mpas_log_write('--- f_ni   = $l',logicArgs=(/f_ni/)  )
+       call mpas_log_write('--- f_nifa = $l',logicArgs=(/f_nifa/))
+       call mpas_log_write('--- f_nwfa = $l',logicArgs=(/f_nwfa/))
+       call mpas_log_write('--- f_nbca = $l',logicArgs=(/f_nbca/))
+       call mpas_log_write(' ')
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F:873]: It looks like these and other print statements have somehow gotten uncommented. Could we comment these out again?

**Review by mgduda — CHANGES_REQUESTED** [review]

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F (hunk 4 above)
Comment (ldfowler58) [reply src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F]: done.

**Review by mgduda — APPROVED** [review]

**Pull request merged**